Update docs (#394)
kauwelab/PolyRiskScore: README.md

# PRSKB

The Polygenic Risk Score Knowledge Base (PRSKB) is a website and command-line interface tool designed to facilitate the calculation of polygenic risk scores using genome-wide association studies from the [NHGRI-EBI Catalog](https://www.ebi.ac.uk/gwas).

## Website

The PRSKB website offers users the ability to calculate polygenic risk scores across multiple traits with minimal effort. The website not only offers the calculator, but additional information and tools to aid in calculations and data comparisons.

Website Link: [https://prs.byu.edu](https://prs.byu.edu)

### Calculate

> The Calculate page of the PRSKB website allows the user to calculate polygenic risk scores for multiple samples across more than 2100 trait/study combinations. Using an input VCF or inputted rsIDs and genotypes, reference genome, super population of the samples, p-value cutoff, and selected studies, polygenic risk scores can be calculated for the inputted samples. Results can be downloaded as a tsv or json file.

Reviewer: As of March 16, 2022, the PRSKB database contains the following data derived from the GWAS Catalog: 250,134 variant associations; 125,433 unique single nucleotide polymorphisms; 20,798 unique study and trait combinations; 10,366 GWA study identifiers; and 3,463 PubMed identifiers. The PRSKB is automatically updated with new studies from the GWAS Catalog monthly.

Reviewer: Instead of "2100 trait/study combinations"

> *Note*: Due to the vast number of studies the PRSKB has access to, it is not feasible to calculate PRS for them all on the website. If you wish to run large amounts of studies in a single run, see our [Command-Line Interface download (CLI)](https://prs.byu.edu/cli_download). Our current limit for number of studies to be run at a time is 500, though we strongly recommend using the CLI for anything more that 50 studies. 

Reviewer: "although we strongly" ; "more than 50 studies"

Reviewer: ```suggestion
*Note*: Due to the vast number of studies the PRSKB has access to, it is not feasible to calculate PRS for them all on the website. If you wish to run large amounts of studies in a single run, see our [Command-Line Interface download (CLI)](https://prs.byu.edu/cli_download). Our current limit for number of studies to be run at a time is 500, although we strongly recommend using the CLI for anything more than 50 studies. 
```

#### Uploading GWAS Summary Statistic Data
In addition to calculating polygenic risk scores using GWA studies from the GWAS Catalog stored in our database, users have the option to upload their own GWAS summary statistics to use in risk score calculations. 

> The GWAS summary statistics file to be uploaded **must** be in the correct format. It should be either a .tsv or a .txt tab separated file. The following columns are required and must be included in the file's header line: Study ID, Trait, RsID, Chromosome, Position, Risk Allele, Odds Ratio, and P-value. If the summary statistics use beta values instead of odds ratios, replace the Odds Ratio column with Beta Coefficients and Beta Units. Additional optional columns that will be included if present are: P-Value Annotation, Beta Annotation, Citation, and Reported Trait. Column order does not matter and there may be extra columns present in the file. Required and optional header names must be exact. Note that if P-value Annotation and/or Beta Annotation are present, then the calculator will separate calculations by those columns. If you do not wish for this to happen, do not include those optional columns.

Reviewer 2: replace the "Odds Ratio" column with two columns: "Beta Coefficients" and "Beta Units."

> If more than one odds ratio exists for an RsID/allele combination in a study, tool will exit. Although we perform strand flipping on GWAS summary statistics data we use from the GWAS Catalog, *we do not currently perform strand flipping on GWAS data uploaded to the website.* Please ensure that your data is presented on the correct strand or use the CLI tool.

Reviewer: "the tool will exit"
"your data are presented"

Reviewer: ```suggestion
If more than one odds ratio exists for an RsID/allele combination in a study, the tool will exit. Although we perform strand flipping on GWAS summary statistics data we use from the GWAS Catalog, *we do not currently perform strand flipping on GWAS data uploaded to the website.* Please ensure that your data are presented on the correct strand or use the CLI tool.
```

> More CLI examples can be found in the README.md file included in the CLI download or on the [readthedocs page](https://polyriskscore.rtfd.io) for the tool.

Reviewer: "Display Ethnicities" image in readthedocs is not rendering for me.

## Output Results

There are two choices for the tsv output results - condensed (default) or full. Additonally, you can choose to output results in JSON format, which contains all the information found in the 'full' format. Explanations of the columns found in the output are given below.

- **Study ID** -- The study identifier assigned by the GWAS Catalog (or the user if they uploaded their own GWAS summary statistics)
- **Reported Trait** -- Trait based on the phenotype being studied, as described by the authors
- **Trait** -- Trait assigned by the GWAS Catalog, standardized from the Experimental Factor Ontology
- **Citation** -- The citation of the study
- **P-Value Annotation** -- Additional information about the p-values
- **Beta Annotation** -- Additional information about the beta values
- **Score Type** -- This indicates if the study used odds ratios or beta values
- **Units (if applicable)** -- This column will contain the beta units if the Score Type is beta. 

> - **SNP Overlap** -- Details the number of SNPs that are in the sample vcf/txt file which are in the study and not excluded from the calculation (see below)

Reviewer 2: "...number of SNPs that are in the <input> vcf/txt file which are <also> in the study and not..."

- **SNPs Excluded Due To Cutoffs** -- Details the number of snps excluded from the study calculation due to p-value cutoff or minor allele frequency threshold
- **Included SNPs** -- The total number of SNPs included in the calculation
- **Used Super Population** -- The super population used for linkage disequillibrium

> #### Columns Only Available In The Full Version

Reviewer: Columns Available Only In The Full Version

Reviewer: ```suggestion
#### Columns Available Only In The Full Version
```

.. code-block:: bash

   ./runPrsCLI.sh -f path/to/file/samples.vcf -o path/to/file/output.tsv -c 0.0005 -r hg19 -p SAS -v

### JSON

This version outputs the results in a json object format. The output automatically contains all the data the full version does and there is no condensed version of the json output. The file will contain a list of json study objects. Each study object will contain the list of samples and their score. 

.. code-block:: bash
    # example output
    [
        {
            "studyID": "GCST001",
            "reportedTrait": "Alzheimer's Disease",
            "trait": "Alzheimer Disease",
            "citation": "First Author et al. 2021",
            "pValueAnnotation": "NA",
            "betaAnnotation": "NA",
            "scoreType": "OR",
            "units (if applicable)": "NA",
            "snpOverlap": 8,
            "excludedSnps": 3,
            "totalSnps": 14,
            "usedSuperPop": "EUR",

>             "samples": [

Reviewer 2: A few questions on how the riskAlleles, variantsWithoutRiskAllele, and variantsInHighLD columns work, relating to this example:
1. With SAMP002 as an example (and the other two as well I guess), if 3 SNPs are excluded from the study, will those SNPs still show up in the variantsWithoutRiskAllele column (or in any of the other columns for that matter)? My expectation is that they would not show up in any of the columns (or in an excluded SNPs column).
2. Will the variantsInHighLD column be the same across all samples, or will it be like SAMP002 and SAMP003 where they are different? My expectation is that they would all be the same.
3. In SAMP003, I expect that the riskAlleles column will not have any overlapping SNPs with the variantsWithoutRiskAllele column. Is that a correct assumption to make?

Author: 1. that is correct
2. using sample-wide clumping, yes they would be the same. I think if you use individual-clumping it could potentially be different.
3. Not necessarily. If someone is heterozygous for the risk allele, then they could have 1 snp in riskAlleles and 1 snp in variantsWithoutRiskAllele. Is it confusing to have it that way? I thought it would be helpful to do it like that to indicate why 2 individuals with the same snps but one is homozygous and one is heterozygous have different scores.

Reviewer 2: 1. 3 of the 14 snps need to not show up in each of the samples in this example
2. SAMP002 needs rs12 and rs13 moved to its variantsInHighLD section.
3. Oh! I see that now. That is not initially intuitive, but makes sense after you have explained it. I think you should keep it this way, but explain above why this overlap occurs sometimes.

Author: Okay I added an explanation and made those changes

                    "protectiveAlleles": "rs1|rs2|rs3",
                    "riskAlleles": "rs4|rs5|rs6|rs7|rs8",
                    "variantsWithoutRiskAllele": "rs9|rs10|rs11|rs14",
                    "variantsInHighLD": "rs12|rs13"
                },
                {
                    "sample": "SAMP002",
                    "polygenicRiskScore": "NF",
                    "protectiveAlleles": "",
                    "riskAlleles": "",
                    "variantsWithoutRiskAllele": "rs1|rs2|rs3|rs4|rs5|rs6|rs7|rs8|rs9|rs10|rs11|rs14",
                    "variantsInHighLD": "rs12|rs13"
                },
                {
                    "sample": "SAMP003",
                    "polygenicRiskScore": "1.63",
                    "protectiveAlleles": "rs1|rs2|rs3",
                    "riskAlleles": "rs4|rs5|rs6|rs7|rs8",
                    "variantsWithoutRiskAllele": "rs1|rs2|rs3|rs4|rs5|rs6|rs7|rs8|rs9|rs10|rs11|rs14",
                    "variantsInHighLD": "rs12|rs13"
                }
            ]
        }
    ]

.. code-block:: bash

   ./runPrsCLI.sh -f path/to/file/samples.vcf -o path/to/file/output.json -c 0.0005 -r hg19 -p SAS

## Citing this work

> Please visit the [Studies](https://prs.byu.edu/studies.htm) page of the website for information on how to cite this tool and GWAS publications used. 

Reviewer: ##License

This work is freely available for academic and not-for-profit use. However, commercial use is regulated by © 2020 Brigham Young University. All rights reserved. For more information about commercial use of this product, please contact Justin Miller, Ph.D. (justin.miller@uky.edu) or Keoni Kauwe, Ph.D. (kauwe@byu.edu).

Author: Is this instead of the Citing this work section or in addition to?

Reviewer: In addition too
```suggestion
Please visit the [Studies](https://prs.byu.edu/studies.htm) page of the website for information on how to cite this tool and GWAS publications used. 

##License

This work is freely available for academic and not-for-profit use. However, commercial use is regulated by © 2020 Brigham Young University. All rights reserved. For more information about commercial use of this product, please contact Justin Miller, Ph.D. ([justin.miller@uky.edu](mailto:justin.miller@uky.edu)) or Keoni Kauwe, Ph.D. ([kauwe@byu.edu](mailto:kauwe@byu.edu)).
```

> Please visit the [Studies](https://prs.byu.edu/studies.htm) page of the website for information on how to cite this tool and GWAS publications used. 

Reviewer: ```suggestion
Please visit the [Studies](https://prs.byu.edu/studies.htm) page of the website for information on how to cite this tool and GWAS publications used. 

##License

This work is freely available for academic and not-for-profit use. However, commercial use is regulated by © 2020 Brigham Young University. All rights reserved. For more information about commercial use of this product, please contact Justin Miller, Ph.D. ([justin.miller@uky.edu](mailto:justin.miller@uky.edu)) or Keoni Kauwe, Ph.D. ([kauwe@byu.edu](mailto:kauwe@byu.edu)).
```
